src/s3poller.ts

```diff
@@ -126,26 +146,45 @@ async function processS3Contents(
   bucket: string,
   contents: any[]
 ) {
-  for (const c of contents) {
-    if (!c.Key) continue
+  // First, fetch and parse all XML files to get their timestamps
+  const filesWithTimestamps = await Promise.all(
+    contents
+      .filter(c => c.Key?.toLowerCase().endsWith('.xml') && !c.Key.includes('batchcomplete'))
+      .map(async (c) => {
+        try {
+          const { Body } = await client.send(
+            new GetObjectCommand({
+              Bucket: bucket,
+              Key: c.Key,
+            })
+          )
+          const xml = await Body?.transformToString()
+          if (xml) {
+            const $ = cheerio.load(xml, { xmlMode: true })
+            const messageTimestamp = $('MessageCreatedDateTime').first().text()
+            return { key: c.Key, xml, messageTimestamp }
+          }
+        } catch (error) {
+          console.error(`Failed to fetch/parse ${c.Key}:`, error)
+        }
+        return null
+      })
+  )
 
-    const lowerKey = c.Key.toLowerCase()
-    if (!lowerKey.endsWith('.xml') || lowerKey.includes('batchcomplete')) {
-      continue
-    }
+  // Sort by messageTimestamp, handling cases where timestamp might be missing
+  const sortedFiles = filesWithTimestamps
+    .filter(Boolean)
+    .sort((a, b) => {
+      const timestampA = a!.messageTimestamp || ''
+      const timestampB = b!.messageTimestamp || ''
+      return timestampA.localeCompare(timestampB)
```

Comment:
Resisting the urge to go down a rabbit hole to determine if locale actually matters when comparing timestamps strings.
Also, feels like we maybe shouldn't do that lexicographically but should parse them to actual numbers and do the comparison with plain ole gt/lt